overlord/devicestate/handlers.go

```diff
@@ -209,6 +209,11 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
 		if err := m.recordSeededSystem(st, whatSeeded); err != nil {
 			return fmt.Errorf("cannot record the seeded system: %v", err)
 		}
+
+		// since this is the most recently seeded system, it should also be the
+		// default recovery system. this is important when coming back from a
+		// factory-reset.
+		st.Set("default-recovery-system", whatSeeded.System)
```

Comment:
this also looks wrong now, please double check all the places this is being set

Reply:
Ugh, sorry I missed these. @bboozzoo thanks for the fixes